### ISSUE TITLE
Istio telemetry causing prometheus to crash

### ISSUE THREAD

**Issue by Siddharthk** [istio/istio#28528]
Istio telemetry is causing prometheus pod in getting evicted. Istio telemetry is giving pod IP and random ports instead of service names.

![image](https://user-images.githubusercontent.com/718854/97954657-22d64880-1dca-11eb-8c9d-cf77c3281b70.png)

[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[x] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Expected behavior

Istio telemetry should only give out service names.

Steps to reproduce the bug:

Sometimes istio telemetry is fluctuating between service name and pod ip thus causing too many services to be scraped by prometheus and ultimately bringing it down. Screenshots below from prometheus and kiali:

![image](https://user-images.githubusercontent.com/718854/97954735-5e711280-1dca-11eb-9bc8-c09e835861c4.png)

The yellow highlighted ones are actually the kubernetes services. IP is one of the pod which the service points to. Also notice random ports being associated with the IP.
image

![image](https://user-images.githubusercontent.com/718854/97954756-692ba780-1dca-11eb-9e03-5101b3e22283.png)

In the above example, the service external-search-kagan-vert-us actually points to the Pod IP 10.21.33.188.
```
external-search-kagan-vert-sa9ahy3f       2/2     Running            0          45h     10.21.33.188   ip-10-21-32-38.ec2.internal    
```
More details here:  kiali/kiali#3354

Version (include the output of istioctl version --remote and kubectl version and helm version if you used Helm)

client version: 1.6.7
control plane version: 1.6.7
data plane version: 1.6.7 (78 proxies)

Client Version: version.Info{Major:"1", Minor:"16+", GitVersion:"v1.16.8-eks-e16311", GitCommit:"e163110a04dcb2f39c3325af96d019b4925419eb", GitTreeState:"clean", BuildDate:"2020-03-27T22:40:13Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"16+", GitVersion:"v1.16.8-eks-e16311", GitCommit:"e163110a04dcb2f39c3325af96d019b4925419eb", GitTreeState:"clean", BuildDate:"2020-03-27T22:37:12Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}

How was Istio installed?

Using istioctl

Environment where bug was observed (cloud vendor, OS, etc)

AWK EKS 1.16.8

**Comment by howardjohn** [comment]
cc @bianpengyuan

**Comment by bianpengyuan** [comment]
The random port does not make sense to me. It looks like source port somehow used in dimension. Could you disable host header fallback follow the document here: https://istio.io/v1.6/docs/tasks/observability/metrics/customize-metrics/. Unfortunately you'd need to deal with EnvoyFilter resource in order to achieve this. This is much easier to configure in 1.7: https://istio.io/latest/docs/tasks/observability/metrics/customize-metrics/

**Comment by Siddharthk** [comment]
@bianpengyuan  I have edited the Envoy filter:
```
disable_host_header_fallback: false
```
But this is still happening. Notice the random ports getting associated with the pod IP in below image.
![image](https://user-images.githubusercontent.com/718854/98355043-a2bd1680-2047-11eb-9455-6c938224d819.png)

**Comment by bianpengyuan** [comment]
@Siddharthk could you set `disable_host_header_fallback` to true? I'd like to rule out host header fallback causing this high cardinality.

**Comment by Siddharthk** [comment]
@bianpengyuan it was true in the first place...changed it back to true but still same issue.
![image](https://user-images.githubusercontent.com/718854/98388585-f8f67d80-2078-11eb-80e2-e20ecbc7ac8f.png)

**Comment by bianpengyuan** [comment]
Could you set it to true at all three places? I believe you flipped the configuration at gateway envoyfilter, could you add the option to sidecar ones?

**Comment by Siddharthk** [comment]
Added in inboundSidecar, outboundSidecar and gateway. Still same issue after restarting prometheus.
![image](https://user-images.githubusercontent.com/718854/98391131-3f99a700-207c-11eb-9c01-23de64de2114.png)

**Comment by bianpengyuan** [comment]
@Siddharthk You'd need to restart sidecars as well, prometheus stats are emitted there and persist for the lifetime of the proxy.

**Comment by Siddharthk** [comment]
@bianpengyuan redployed the apps and monitored for 5 mins...I don't see this happening now. Is this the solution?

**Comment by Siddharthk** [comment]
@bianpengyuan let me know what needs to be done next

**Comment by bianpengyuan** [comment]
Yeah, this is the solution. Although for some cases where destination service cannot be determined, for example, client sends request to some unknown host, destination_service label won't be filled with the host head if `disable_host_header_fallback` is true. If that is not a concern, it makes sense to just disable host header fallback globally.

**Comment by Siddharthk** [comment]
I am just wondering if its me only or other people are also facing this issue since this setting is not there by default.

**Comment by bianpengyuan** [comment]
Most of the time, host header is bounded for in mesh requests, so that's why we enabled it by default.

Not sure why it uses random ports. It reads that from request host header, so totally depends on how app behaves.

**Comment by douglas-reid** [comment]
Can this issue be closed now?  It seems like it has been resolved.

**Comment by bianpengyuan** [comment]
Closing this now, @Siddharthk please reopen if needed. Thanks!
